From f223f3ccade4563ac1e6d1f375bf947d1221937d Mon Sep 17 00:00:00 2001
From: Zhu Lingshan <lszhu@suse.com>
Date: Thu, 25 Aug 2016 16:09:28 +0800
Subject: [PATCH] Add a missed complete in iscsit_close_connection
References: bsc#992555, bsc#987805
Patch-mainline: Not yet, will submit later

Add a missed complete operation for session_wait_comp in
fall_back_to_erl routine. Without this complete operation,
iscsit_stop_session() would stuck, left D threads under
stress tests

Signed-off-by: Zhu Lingshan <lszhu@suse.com>
Reviewed-by: Lee Duncan <lduncan@suse.com>
---
 drivers/target/iscsi/iscsi_target.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/target/iscsi/iscsi_target.c b/drivers/target/iscsi/iscsi_target.c
index 72204fb..924a9fe 100644
--- a/drivers/target/iscsi/iscsi_target.c
+++ b/drivers/target/iscsi/iscsi_target.c
@@ -4433,6 +4433,8 @@ int iscsit_close_connection(
 	     atomic_read(&sess->session_fall_back_to_erl0)) {
 		spin_unlock_bh(&sess->conn_lock);
 		target_put_session(sess->se_sess);
+                if (atomic_read(&sess->sleep_on_sess_wait_comp))
+                        complete(&sess->session_wait_comp);
 
 		return 0;
 	} else if (atomic_read(&sess->session_logout)) {
-- 
2.6.6

